### ISSUE TITLE
Limited number of synchronized books

### ISSUE THREAD

**Issue by alfamatteo92** [fsantini/KoboCloud#36]
Hello, I used your script on my Kobo Clara and I am able to synchronized my books on the cloud, but there is a problem. The number of the books that I can synchronized is limited. I have a folder with all my books (more than 100) on dropbox and the same folder also on google drive. With Gdrive I am able to synchronize 41 books while with dropbox only 30. How can I solve this problem? I tried several times, also rebooting the device but nope... there are only the same books and it didn't import the other ones. How can I fix and synchronize all the books? Thank you very much for your work

This is the log
[get.log](https://github.com/fsantini/KoboCloud/files/4948641/get.log)

**Comment by ScottESanDiego** [comment]
I had a similar issue with both those same storage providers (and there's a warning in the readme about potential issues syncing large numbers with Google Drive), so you're not alone.

FWIW, I used the `rclone` method (see the PR), and sync'ed  162 books from DropBox w/o issue.  If you _really_ need to sync all your books, you could go that route. (Not sure if @fsantini is going to merge it, or if there's something else waiting to happen.)

**Comment by ionontelodico** [comment]
Hi & Ciao
excuse me if i post here my problem too 
[LOG_.txt](https://github.com/fsantini/KoboCloud/files/5057170/LOG_.txt)
 (as i did on mobilereadforum)...
...i still have problems downloading my ebook from google drive to my Kobo Glo HD...
This is my kobocloudrc file:

```
# Add your URLs to this file
# Remove the # from the following line to uninstall KoboCloud
#UNINSTALL
# Google Drive
https://drive.google.com/drive/folders/My_Shared_Folder
enter character
```


My experience:
The cloud downloading worked the first time, but not completely... i have 74 ebooks, and my kobo glo hd downloaded only 41; every time i switch on my kobo, it seems to me it stucks on 6% of downloading...and still i have only 41 books
You can see my Kobo Log in the attachment...
I downloaded rclone, but really i don't know how can it works on the kobo...
Can you help me? Thanks & Grazie

**Comment by ScottESanDiego** [comment]
Do the instructions I put in https://github.com/fsantini/KoboCloud/pull/35 along with https://rclone.org/docs/ help with how to use rclone?

**Comment by ionontelodico** [comment]
Thanks a lot for the answer...but still my kobo stucks on 41 books!
I followed your tips, but really don't understood the 3rd point
`3 Add a line to kobocloudrc that looks like RCLONE Foo (e.g., RCLONE ScottE_DropBox:eBooks)`
What is a Foo? i added this line in my kobocloudrc file:
`RCLONE ebook:`
(my remote is named "ebook"...)
...but it's not working..
What am i doing wrong?
Thanks & ciao

**Comment by ScottESanDiego** [comment]
@ionontelodico rclone needs a "config file" which is created when you run the `rclone config` command.

What I did is download `rclone` to my desktop system, run `rclone config` and create a new config file as described by [the rclone docs](https://rclone.org/commands/rclone_config_create/) . 

Like, if you run through the `rclone config` steps on your desktop and make a new config for DropBox named 'ebook', you should now have a `.rclone.conf` file on your desktop with contents that start with `[ebook]`.  (In the 3rd point example I made, my config was named `ScottE_Dropbox` and it had a subdirectory `eBooks`, hence `ScottE_Dropbox:eBooks`).

Copy that `.rclone.conf` file from your desktop system to the `kobo_rclone.conf` in the `kobocloud` folder on your Kobo.

Finally, edit `kobocloudrc` to reflect the name of the share you want to use.  For your case, probably something like `RCLONE ebook:` would be the right thing, and sounds like what you did already.

Can you confirm that `ebook` matches what's in the `kobo_rclone.conf` file on the Kobo?

**Comment by ionontelodico** [comment]
Thanks a lot @ScottESanDiego ...rclone worked,  but i had to migrate from gdrive to dropbox...and finally my KoboGlo downloaded all my stored ebooks!
The good is that kobocloudrc didn't duplicate ebooks in this migration...really sync!
Thanks really so much! 
:-)

**Comment by ScottESanDiego** [comment]
Why do you have to migrate?  You should be able to configure `rclone` to work with Google Drive (I use it with Google Drive and Dropbox).

Anyway, glad it worked!  If you want to contribute anything to that PR (like better instructions than what I had), that'd be useful.  Hoping we can get @fsantini to merge it, so I don't have to maintain my changes. :-)

**Comment by ionontelodico** [comment]
`Why do you have to migrate? `
... Well... Gdrive always stucks on 6%...
Migrating, sync was quite immediate...
 i added 2 more ebooks.. Kobo started... About after 2 minutes went to sync... I want to test when (after the device's boot) exactly kobo start the sync...
P. S.: i think your tips are more clear... My english is maccaroni english!

**Comment by ionontelodico** [comment]
Hi again...
Again,my Kobo Glo hd doesn't sync with my dropbox shared folder... Reallly, i don't understand...
I have 79 ebooks in my folder, and my kobo loads only 50...
Updated mu rclone token...but nothing...
This is my rclone.conf

```
[ebook_drop]
type = dropbox
token = {"access_token":"ZhjFG4cXXrIAAAAAAAAAATo_GavyUC7LyrrB_G9QgtuQf6tetpNYaxx-rtB0Zztl","token_type":"bearer","expiry":"0001-01-01T00:00:00Z"}
```

This is my koboclourc
```
https://www.dropbox.com/sh/b3303jlbwk63bxh/AABywGY3Eg_RXECykvDdnx1Ra?dl=0
https://www.dropbox.com/sh/b3303jlbwk63bxh/AABywGY3Eg_RXECykvDdnx1Ra?dl=0
RCLONE ebook_drop:
# Add your URLs to this file
# Remove the # from the following line to uninstall KoboCloud
#UNINSTALL
```
Where do i go wrong?

**Comment by ScottESanDiego** [comment]
Doesn't the `[ebook_drop]` section also need a `client_id` and `client_secret` ?

The `kobocloudrc` needs _either_ the HTTPS URL _or_ the RCLONE line.  Not both.  I'd comment out the HTTPS line and see what happens.

**Comment by ionontelodico** [comment]
First: Thanks a lot!
Then...let me understand: i have to delete the _https_ line and keep this quote
`RCLONE ebook_drop:`
?
...and where can i find the` client id `and `client_secret` you said?
Thanks again! :-)

**Comment by ScottESanDiego** [comment]
Correct. You should _only_ have `RCLONE ebook_drop:` in your kobocloudrc file (other than comments of course).

My `.rclone.conf` has those `client_id` and `client_secret` lines in it for my DropBox entry:

```
[ScottESanDiego_DropBox]
type = dropbox
token = {"access_token":"<blahblahblah>","token_type":"bearer","expiry":"0001-01-01T00:00:00Z"}
client_id = blahblahblah
client_secret = blahblahblah
```

Not sure why your `rclone config` didn't add those.  I mean, I guess if it didn't add them they they aren't needed...just it looks different than mine. :-)

**Comment by ionontelodico** [comment]
Nothing...i deleted the http line...even added 
```
client_id = ZhjFG4cXXrIAAAAAAAAAATo_GavyUC7LyrrB_G9QgtuQf6tetpNYaxx-rtB0Zztl
client_secret = ZhjFG4cXXrIAAAAAAAAAATo_GavyUC7LyrrB_G9QgtuQf6tetpNYaxx-rtB0Zztl
```
but nothing...
I think ...it's a strange behaviour: my kobo loads only a limited number of ebooks... 
Thanks a lot, however!
:-)
